### PR TITLE
Feature/SE-2385.InlinereconciliationNb

### PR DESCRIPTION
**Pull request by DavidWassellFinbourne**
# Pull Request Checklist

- [x] Changes follow the [style guide](https://github.com/finbourne/sample-notebooks/blob/master/docs/FINBOURNE%20notebook%20style%20guide.ipynb)
- [x] Tests pass
- [x] Raised the PR against the `develop` branch

# Description of the PR

PR for Inline Reconciliation NB.

Original PR here:

https://github.com/finbourne/sample-notebooks/pull/515

### CODE REVIEW

File: examples/use-cases/post-trade-processing/Transactions with Trade To Portfolio Rate.ipynb [hunk 1]
```diff
@@ -0,0 +1,680 @@
+{
```
Comment (Tomwp) [inline examples/use-cases/post-trade-processing/Transactions with Trade To Portfolio Rate.ipynb:1]: <p>You've mentioned it below, but I would state here that there's two ways to set the Trade To Portfolio Rate:</p><ol><li>Via pulling data out of the quote store</li><li>Via setting it on the transaction</li></ol><p> </p><p>You should also mention the fact that TTPR is used to calculate a positions cost basis in portfolio currency. I don't think the cost of the 'trade' is as important, but you can probably keep that there.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/557/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (Tomwp) [inline examples/use-cases/post-trade-processing/Transactions with Trade To Portfolio Rate.ipynb:1]: <p>Might be worth calling the portfolio "TTPR_NB_Portfolio"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/557/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

File: examples/use-cases/ibor/Inline Reconciliation.ipynb [hunk 2]
```diff
@@ -0,0 +1,598 @@
+{
```
Comment (Tomwp) [inline examples/use-cases/ibor/Inline Reconciliation.ipynb:1]: <ol><li><em>"reconciliation of two of valuations"</em> - confusing wording. I would maybe say: "This notebook shows you have to complete an inline reconciliation of two valuations via weighted instruments using two separate pricing sources."</li></ol>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/557/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (Tomwp) [inline examples/use-cases/ibor/Inline Reconciliation.ipynb:1]: <p>Add a sentence to the end of the first line of this section that says something like: "We'll demonstrate this by creating:". You may need to rephrase the bullets (or rephrase the lead on sentence). It's slightly confusing when you just jump to listing those 4 bullets.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/557/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (Tomwp) [inline examples/use-cases/ibor/Inline Reconciliation.ipynb:1]: <p>What's the purpose of line 58: <strong><em>if i in range(0, 5):</em></strong> ?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/557/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (Tomwp) [inline examples/use-cases/ibor/Inline Reconciliation.ipynb:1]: <ol><li>Regarding the following sentence: "<em>The output shows a difference in PV for the 'client_internal_2' and 'client_internal_4' weighted instruments which matches up with the price difference between the 2 sources referenced in the recipes". </em>I would actually add that after the table output and say '<em>The above output...</em>'. I think this is clearer.</li><li>Your last sentence - "<em>The only data written to LUSID...</em>" is a bit of a run-on sentence. I would reword it.</li></ol>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/557/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

File: examples/use-cases/post-trade-processing/Transactions with Trade To Portfolio Rate.ipynb (hunk 1 above)
Comment (DavidWassellFinbourne) [reply examples/use-cases/post-trade-processing/Transactions with Trade To Portfolio Rate.ipynb:1]: <p>Updated in next pull.</p>

Comment (DavidWassellFinbourne) [reply examples/use-cases/post-trade-processing/Transactions with Trade To Portfolio Rate.ipynb:1]: <p>Updated in next push</p>

File: examples/use-cases/ibor/Inline Reconciliation.ipynb (hunk 2 above)
Comment (DavidWassellFinbourne) [reply examples/use-cases/ibor/Inline Reconciliation.ipynb:1]: <p>Updated</p>

Comment (DavidWassellFinbourne) [reply examples/use-cases/ibor/Inline Reconciliation.ipynb:1]: <p>Updated</p>

Comment (rizwansaeed) [reply examples/use-cases/ibor/Inline Reconciliation.ipynb:1]: <p>"where you don't want to flood your LUSID environment with excess data."</p><p> </p><p>i think this is unnecessary</p>

Comment (rizwansaeed) [inline examples/use-cases/ibor/Inline Reconciliation.ipynb:1]: <p>i would change "same 5 client internals" &gt; "same 5 instruments"</p><p> </p><p>need to define what "weighted instruments" means</p><p> </p><p>i think the definition of inline could be a bit clearer, i think what you mean is more along the lines of them being created on the fly and not persisted in LUSID</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/557/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (rizwansaeed) [inline examples/use-cases/ibor/Inline Reconciliation.ipynb:1]: <p>these can be collapsed into 1 upsert call passing in 2 quotes</p><p> </p><p>are there any KB docs explaining what WeightedInstruments are? its the first time i've seen them!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/557/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (rizwansaeed) [inline examples/use-cases/ibor/Inline Reconciliation.ipynb:1]: <p>typo seperate &gt; separate</p><p> </p><p> </p><p>These recipes determine that equity quotes (mid) loaded into the quotes store in scopes "pricing_scope_a" or "pricing_scope_b" (depending on which is used) are to be used for valuations. &gt; </p><p>These recipes describe how to resolve the quotes loaded into the quotes store for use in the valuations. </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/557/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (rizwansaeed) [inline examples/use-cases/ibor/Inline Reconciliation.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #26.                default_instrument_code_type="Isin",
</pre><p>do you need this section? you're not loading anything with Isins (from what i can see)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/557/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (rizwansaeed) [inline examples/use-cases/ibor/Inline Reconciliation.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #52.                    key="Equity.ClientInternal.*",
</pre><p> </p><p>add a comment explaining this line is resolving based on quotes uploaded with <code>ClientInternal</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/557/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (rizwansaeed) [inline examples/use-cases/ibor/Inline Reconciliation.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #54.                    data_scope=pricing_scope_b,
</pre><p>in this scope</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/557/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (rizwansaeed) [inline examples/use-cases/ibor/Inline Reconciliation.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #57.                    quote_interval="1D.0D"
</pre><p>with a lookback of 1 day from the effectiveDate</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/557/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (rizwansaeed) [inline examples/use-cases/ibor/Inline Reconciliation.ipynb:1]: <p>is <code>InstrumentTag</code> the instrument id (LUID) or something else? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/557/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

**Comment by hanialsibai** [comment]
Closing as similar to  https://github.com/finbourne/sample-notebooks/blob/master/examples/use-cases/valuation/Inline%20Valuations%20with%20Inferred%20FX%20Rates.ipynb
